impl/src/main/java/org/ehcache/spi/event/DefaultCacheEventListenerFactory.java

```diff
@@ -42,4 +43,11 @@ public DefaultCacheEventListenerFactory() {
   public void releaseEventListener(CacheEventListener<?, ?> cacheEventListener) {
     //noop
   }
+
+  @Override
+  public void start(ServiceConfiguration<?> config, ServiceProvider serviceProvider) {
```

Comment:
You can't just override the start method from `ClassInstanceProvider` without calling super ...
And in the case of services extending `ClassInstanceProvider` the check needs to be a bit smarter.